viam-cartographer_internal_test.go

```diff
@@ -0,0 +1,332 @@
+// Package builtin implements simultaneous localization and mapping.
+// This is an Experimental package.
+package viamcartographer
+
+import (
+	"bytes"
+	"context"
+	"testing"
+
+	"github.com/golang/geo/r3"
+	"github.com/pkg/errors"
+	commonv1 "go.viam.com/api/common/v1"
+	v1 "go.viam.com/api/service/slam/v1"
+	"go.viam.com/rdk/services/slam"
+	"go.viam.com/rdk/spatialmath"
+	"go.viam.com/rdk/testutils/inject"
+	"go.viam.com/test"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+var (
+	chunkSizePointCloudMap     int
+	chunkSizeInternalState     int
+	inputComponentRef          string
+	expectedPointCloudMapBytes []byte
+	expectedInternalStateBytes []byte
+	inputQuat                  map[string]interface{}
+	inputPose                  commonv1.Pose
+)
+
+type pointCloudClientMock struct {
+	grpc.ClientStream
+	reader           *bytes.Reader
+	clientBufferSize int
+}
+
+func makePointCloudClientMock(b *[]byte, clientBufSize *int) *pointCloudClientMock {
+	return &pointCloudClientMock{reader: bytes.NewReader(*b), clientBufferSize: *clientBufSize}
+}
+
+func (m *pointCloudClientMock) Recv() (*v1.GetPointCloudMapResponse, error) {
+	clientBuffer := make([]byte, m.clientBufferSize)
+	n, err := m.reader.Read(clientBuffer)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := &v1.GetPointCloudMapResponse{
+		PointCloudPcdChunk: clientBuffer[:n],
+	}
+	return resp, nil
+}
+
+type internalStateClientMock struct {
+	grpc.ClientStream
+	reader           *bytes.Reader
+	clientBufferSize int
+}
+
+func makeInternalStateClientMock(b *[]byte, clientBufSize *int) *internalStateClientMock {
+	return &internalStateClientMock{reader: bytes.NewReader(*b), clientBufferSize: *clientBufSize}
+}
+
+func (m *internalStateClientMock) Recv() (*v1.GetInternalStateResponse, error) {
+	clientBuffer := make([]byte, m.clientBufferSize)
+	n, err := m.reader.Read(clientBuffer)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := &v1.GetInternalStateResponse{
+		InternalStateChunk: clientBuffer[:n],
+	}
+	return resp, nil
+}
+
+// func makeGetPositionResponse(pose *spatialmath.Pose, componentReference *string) (*v1.GetPositionResponse, error) {
```

Comment:
[note] Another option for building the GetPositionResponse but provides less flexibility in the failure tests we can perform

Reply:
I think we can remove this option